src/top/run/metrics_util.py

```diff
@@ -0,0 +1,280 @@
+"""
+Util classes for computing evaluation metrics.(HitMiss, Accuracy, AveragePrecision, IoU)
+Reference: https://github.com/google-research-datasets/Objectron/blob/master/objectron/dataset/metrics.py
+"""
+
+import numpy as np
+import scipy.spatial as sp
+import src.top.run.box_generator as Box
+
+
+class HitMiss(object):
+    """Class for recording hits and misses of detection results."""
+
+    def __init__(self, thresholds):
+        self.thresholds = thresholds
+        self.size = thresholds.shape[0]
+        self.hit = np.zeros(self.size, dtype=np.float)
+        self.miss = np.zeros(self.size, dtype=np.float)
+
+    def reset(self):
+        self.hit = np.zeros(self.size, dtype=np.float)
+        self.miss = np.zeros(self.size, dtype=np.float)
+
+    def record_hit_miss(self, metric, greater=True):
+        """Records the hit or miss for the object based on the metric threshold."""
+        # NOTE(Jiyong): greater True -> find greater than threshold / False -> find worse than threshold
+
+        for i in range(self.size):
+            threshold = self.thresholds[i]
+            hit = (greater and metric >= threshold) or ((not greater) and metric <= threshold)
+            if hit:
+                self.hit[i] += 1
+            else:
+                self.miss[i] += 1
+
+
+class Accuracy(object):
+    """Class for accuracy metric."""
+
+    def __init__(self):
+        self._errors = []
+        self.acc = []
+
+    def add_error(self, error):
+        """Adds an error."""
+        self._errors.append(error)
+
+    def compute_accuracy(self, thresh=0.1):
+        """Computes accuracy for a given threshold."""
+        if not self._errors:
+            return 0
+        return len(np.where(np.array(self._errors) <= thresh)[0]) * 100. / len(self._errors)
```

Comment:
If we're going to use percentages this choice should be documented. Since we're multiplying by 100 I'm assuming we're using percentages here.